File: src/main/java/org/oewntk/xml/in/XmlNames.java
package org.oewntk.xml.in;

/**
 * Tags and attributes in XML files
 *
 * @author Bernard Bou
 */
public class XmlNames
{
	private XmlNames()
	{
	}

	/**
	 * DC XML namespace
	 */
	protected static final String NS_DC = "https://globalwordnet.github.io/schemas/dc/";

	// elements

	static final String LEXICALRESOURCE_TAG = "LexicalResource";

	static final String LEXICON_TAG = "Lexicon";

	static final String LEXICALENTRY_TAG = "LexicalEntry";

	static final String SYNSET_TAG = "Synset";

	static final String DEFINITION_TAG = "Definition";

	static final String EXAMPLE_TAG = "Example";

	static final String SYNSETRELATION_TAG = "SynsetRelation";

	static final String WIKIDATA_TAG = "WikiData";

	static final String LEMMA_TAG = "Lemma";

	static final String SENSE_TAG = "Sense";

	static final String SENSERELATION_TAG = "SenseRelation";

	static final String FORM_TAG = "Form";

	static final String PRONUNCIATION_TAG = "Pronunciation";

	static final String SYNTACTICBEHAVIOUR_TAG = "SyntacticBehaviour";

	// attributes

	static final String ID_ATTR = "id";

	static final String N_ATTR = "n";

	static final String POS_ATTR = "partOfSpeech";

	static final String WRITTENFORM_ATTR = "writtenForm";

	static final String VARIETY_ATTR = "variety";

	static final String SYNSET_ATTR = "synset";

	static final String MEMBERS_ATTR = "members";

	static final String TARGET_ATTR = "target";

	static final String RELTYPE_ATTR = "relType";

	static final String LEXFILE_ATTR = "lexfile";

	static final String ADJPOSITION_ATTR = "adjposition";

	static final String VERBFRAMES_ATTR = "subcat";

	static final String VERBFRAME_ATTR = "subcategorizationFrame";
}

File: src/main/java/org/oewntk/xml/in/XmlNames.kt
/*
 * Copyright (c) 2024. Bernard Bou.
 */

package org.oewntk.xml.`in`

/**
 * Tags and attributes in XML files
 *
 * @author Bernard Bou
 */
@Suppress("unused")
object XmlNames {

    /**
     * DC XML namespace
     */
    @Suppress("unused")
    internal const val NS_DC: String = "https://globalwordnet.github.io/schemas/dc/"

    // elements
    const val LEXICALRESOURCE_TAG: String = "LexicalResource"

    const val LEXICON_TAG: String = "Lexicon"

    const val LEXICALENTRY_TAG: String = "LexicalEntry"

    const val SYNSET_TAG: String = "Synset"

    const val DEFINITION_TAG: String = "Definition"

    const val EXAMPLE_TAG: String = "Example"

    const val USAGE_TAG: String = "Usage"

    const val SYNSETRELATION_TAG: String = "SynsetRelation"

    const val WIKIDATA_TAG: String = "WikiData"

    const val LEMMA_TAG: String = "Lemma"

    const val SENSE_TAG: String = "Sense"

    const val SENSERELATION_TAG: String = "SenseRelation"

    const val FORM_TAG: String = "Form"

    const val PRONUNCIATION_TAG: String = "Pronunciation"

    const val SYNTACTICBEHAVIOUR_TAG: String = "SyntacticBehaviour"

    // attributes
    const val ID_ATTR: String = "id"

    const val N_ATTR: String = "n"

    const val POS_ATTR: String = "partOfSpeech"

    const val WRITTENFORM_ATTR: String = "writtenForm"

    const val VARIETY_ATTR: String = "variety"

    const val SYNSET_ATTR: String = "synset"

    const val MEMBERS_ATTR: String = "members"

    const val TARGET_ATTR: String = "target"

    const val RELTYPE_ATTR: String = "relType"

    const val LEXFILE_ATTR: String = "lexfile"

    const val ADJPOSITION_ATTR: String = "adjposition"

    const val VERBFRAMES_ATTR: String = "subcat"

    const val VERBFRAME_ATTR: String = "subcategorizationFrame"

    const val EXAMPLE_SOURCE_ATTR: String = "source"

    const val ILI_ATTR: String = "ili"
}
